v2/test/utils/helpers.go

```diff
@@ -0,0 +1,77 @@
+package utils
+
+import (
+	"crypto/rand"
+	"encoding/json"
+	"fmt"
+	"log"
+	"math/big"
+	"net"
+)
+
+func GetIpAddress() string {
+	conn, err := net.Dial("udp", "8.8.8.8:80")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer conn.Close()
+
+	localAddr := conn.LocalAddr().(*net.UDPAddr)
+
+	return localAddr.IP.String()
+}
```

Comment:
may I present you to our inhouse solution
https://github.com/mongodb/go-client-mongodb-atlas/blob/d8b9275b859d6e402474a2acc292670e57435932/mongodbatlas/ip_info.go#L41-L42